### PR TITLE
SSL improvements

### PR DESCRIPTION
**Pull request by timhawes**
Documentation fixed for wrap_socket (indentation to place it under the correct class, and added the self argument).

SSLContext.wrap_socket now accepts server_hostname=None (the default value). This previously returned an error (TypeError: Can't convert 'NoneType' object to str implicitly).

Implemented the server_hostname argument which was previously ignored. This is done by giving the SSLSocket it's own copy of the ESP-TLS config structure so that we can set the common_name value within it.

Implemented the check_hostname property, load_verify_locations() and set_default_verify_paths() methods on SSLContext. These are based on the documented CPython interfaces.

### CODE REVIEW

**Comment by ladyada** [comment]
thanks! :)

**Review by gregoberfield — APPROVED** [review]
Looks good to me, adding in the SSL Context functions.

File: ports/espressif/common-hal/ssl/SSLContext.c [hunk 1]
```diff
@@ -55,6 +62,36 @@ ssl_sslsocket_obj_t *common_hal_ssl_sslcontext_wrap_socket(ssl_sslcontext_obj_t
...
+    self->ssl_config.cacert_buf = NULL;
+    self->ssl_config.cacert_bytes = 0;
+}
+
+bool common_hal_ssl_sslcontext_get_check_hostname(ssl_sslcontext_obj_t *self) {
+    if (self->ssl_config.skip_common_name) {
+        return 0;
+    } else {
+        return 1;
+    }
```
Comment (tannewt) [inline ports/espressif/common-hal/ssl/SSLContext.c:88]:
```suggestion
    return !self->ssl_config.skip_common_name;
```

File: ports/espressif/common-hal/ssl/SSLContext.c [hunk 2]
```diff
@@ -55,6 +62,36 @@ ssl_sslsocket_obj_t *common_hal_ssl_sslcontext_wrap_socket(ssl_sslcontext_obj_t
...
+}
+
+void common_hal_ssl_sslcontext_set_check_hostname(ssl_sslcontext_obj_t *self, bool value) {
+    if (value) {
+        self->ssl_config.skip_common_name = 0;
+    } else {
+        self->ssl_config.skip_common_name = 1;
+    }
```
Comment (tannewt) [inline ports/espressif/common-hal/ssl/SSLContext.c:96]:
```suggestion
    self->ssl_config.skip_common_name = !value;
```

**Review by tannewt — CHANGES_REQUESTED** [review]
Two minor suggestions. Thanks for following the CPython API!

**Review by tannewt — APPROVED** [review]
Thank you!

**Pull request merged**